src/crimson/common/operation.h

```diff
@@ -523,29 +558,29 @@ class PipelineHandle {
   template <typename OpT, typename T>
   seastar::future<>
   enter(T &stage, typename T::BlockingEvent::template Trigger<OpT>&& t) {
-    assert(stage.core == seastar::this_shard_id());
-    auto wait_fut = wait_barrier();
-    if (wait_fut.has_value()) {
-      return wait_fut.value().then([this, &stage, t=std::move(t)] () mutable {
-        auto fut = t.maybe_record_blocking(stage.enter(t), stage);
-        exit();
```

Comment:
@athanatos  I probably understand the issue here -- `exit()` should be moved after `fut` is resolved.

Specifically, the shared-mutex from concurrent `get-obc` phase should be released (by `exit()`) **after** the request has entered the next exclusive `process` phase.

Reply:
Ah, that makes sense.  Approved the PR.